### PR TITLE
[tests-only][full-ci] test: Add acceptance test for empty CallToAction notification translation fix

### PR DESCRIPTION
**Pull request by anon-pradip**
## Description

Adds an acceptance test (issue #11908) that an `es` recipient's unshare email is correctly translated and contains no leaked PO file headers, and syncs the share-removed strings into the `tests/config/translations` fixture used by CI
 
**PO (Portable Object)** - the gettext translation file format (.po). Contains msgid/msgstr pairs. The empty msgid "" block (the PO header) holds metadata like Project-Id-Version, MIME-Version, Language, etc.
 
#### Why `tests/config/translations` was updated
 
The acceptance test pipeline runs oCIS with `OCIS_TRANSLATION_PATH=tests/config/translations` (see `tests/acceptance/run-github.py`), so the notifications and userlog services load translations from that test fixture instead of the catalogs embedded in the binary.
 
- `notifications.po` (`es`/`de`): the fixture only contained the share-created strings - the share-removed (unshare) entries were missing. Since gotext returns the untranslated msgid (English) for missing keys, the unshare email stayed English in CI even for an `es` recipient. Added the missing `ShareRemoved` entries (subject, greeting and body), copied verbatim from the embedded catalogs at `services/notifications/pkg/email/l10n/locale/{es,de}/LC_MESSAGES/notifications.po`.
- `userlog.po` (`es`/`de`): added the unshare in-app notification entries (`Resource unshared` subject and message), copied verbatim from `services/userlog/pkg/service/l10n/locale/{es,de}/LC_MESSAGES/userlog.po`, so the in-app notification check in the scenario also passes in CI.
 
The scenario now asserts the translated (Spanish) unshare email body, which passes locally (embedded translations) and in CI (updated fixtures), and verifies the email is free of raw PO header markers (issue #11908).

Related: owncloud/ocis#11979, owncloud/ocis#11908
Part of: https://github.com/owncloud/ocis/issues/12472

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by kw-security** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.eu.snyk.io/org/owncloud-infinite-scale/pr-checks/d2b7058a-cb0a-4e55-aa47-164a279540df?refs=473ac31800dce7af46349738ab496376d0598eb6&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.eu.snyk.io/org/owncloud-infinite-scale/pr-checks/d2b7058a-cb0a-4e55-aa47-164a279540df/license?refs=473ac31800dce7af46349738ab496376d0598eb6&source=prChecksComment) |
| :white_check_mark: | Code Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.eu.snyk.io/org/owncloud-infinite-scale/project/19723732-82b3-4add-aae1-42dd3057212c/pr-check/42e65572-0a7e-42b3-b8b7-133d2bf91d25?refs=473ac31800dce7af46349738ab496376d0598eb6&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: tests/acceptance/bootstrap/EmailContext.php [hunk 1]
```diff
@@ -255,4 +255,49 @@ public function userShouldHaveReceivedTheFollowingGroupedEmail(
...
+		string $user,
+		string $unexpectedEmailBodyContent,
+	): void {
+		$address = $this->featureContext->getEmailAddressForUser($user);
+		$actualEmailBodyContent = $this->getBodyOfLastEmail($address);
+		Assert::assertStringNotContainsString(
+			$unexpectedEmailBodyContent,
+			$actualEmailBodyContent,
+			"The email address '$address' should NOT have received an"
+			. "email with the body containing $unexpectedEmailBodyContent
```
Comment (nabim777) [inline tests/acceptance/bootstrap/EmailContext.php:299]:
```suggestion
			. " email with the body containing $unexpectedEmailBodyContent
```

File: tests/acceptance/features/apiNotification/notification.feature [hunk 2]
```diff
@@ -414,3 +414,23 @@ Feature: Notification
...
+      | resource        | textfile1.txt |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Viewer        |
+    When user "Alice" removes the access of user "Brian" from resource "textfile1.txt" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    And user "Brian" should have received an email from user "Alice" that does not contain
+      """
+      Project-Id-Version
```
Comment (nabim777) [inline tests/acceptance/features/apiNotification/notification.feature:431]:
```suggestion
      Project-Id-Version
      Language: es
      Content-Type:
```
I think these things also need to be included

Comment (anon-pradip) [reply tests/acceptance/features/apiNotification/notification.feature]: umm that needs multiple steps of assertions or we need to change the step def. But using `Project-Id-Version` will be sufficient IMO :thinking:

Comment (nabim777) [reply tests/acceptance/features/apiNotification/notification.feature]: The headers can vary. For example, they could be `Content-Type:`, `Language: es`, or `Project-Id-Version:`. In some case `Project-Id-Version` may not appear, so it would be better to check for other PO headers as well.

Comment (anon-pradip) [reply tests/acceptance/features/apiNotification/notification.feature]: now asserted with only positive email body so if any headers kind of things anyhow inserted in the mail body, the test will fail.

File: tests/acceptance/features/apiNotification/notification.feature [hunk 3]
```diff
@@ -414,3 +414,30 @@ Feature: Notification
       | resource   |
       | HelloWorld |
       | text.txt   |
+
+  @issue-11908 @email
+  Scenario: recipient does not get PO file headers in unshare email for translated locale
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiNotification/notification.feature]:
```suggestion
  Scenario: recipient should not get PO file headers on removing access from resource for translated locale
```

File: tests/acceptance/features/apiNotification/notification.feature [hunk 4]
```diff
@@ -414,3 +414,30 @@ Feature: Notification
...
+      | resource        | textfile1.txt |
+      | space           | Personal      |
+      | sharee          | Brian         |
+      | shareType       | user          |
+      | permissionsRole | Viewer        |
+    When user "Alice" removes the access of user "Brian" from resource "textfile1.txt" of space "Personal" using the Graph API
+    Then the HTTP status code should be "204"
+    And user "Brian" should get a notification with subject "Recurso no compartido" and message:
+      | message |
+      | Alice Hansen dejó de compartir textfile1.txt contigo |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiNotification/notification.feature]:
```suggestion
      | message                                              |
      | Alice Hansen dejó de compartir textfile1.txt contigo |
```

**Review by PrajwolAmatya — APPROVED** [review]
